datajs/src/browser-utils/utils.js

```diff
@@ -26,29 +29,40 @@ export async function toNodeStream(reader, size) {
     // Decode the current chunk to string and prepend the last string
     const string = `${lastString}${decoder.decode(value)}`
 
+    chunkText += string
+
     // Extract lines from chunk
     const lines = string.split(/\r\n|[\r\n]/g)
 
     // Save last line, as it might be incomplete
     lastString = lines.pop() || ''
 
     for (const line of lines) {
-      if (lineCounter === size) break
+      if (lineCounter === size) {
+        reader.cancel()
+        break
+      }
       // Write each string line to our nodejs stream
       nodeStream.push(line + '\r\n')
       lineCounter++
     }
   }
 
   nodeStream.push(null)
+
+  //return a chunk of the file. Chunk is used when parsing large files in CSV modeule
+  if (return_chunk) {
+    return chunkText
+  }
+
   return nodeStream
 }
 
 export function isFileFromBrowser(file) {
   return file instanceof File
 }
 
-export function readChunked(file, chunkCallback, endCallback){
+export function readChunked(file, chunkCallback, endCallback) {
```

Comment:
OK, we have missed JSDoc here previously, so add them to make the code improved. Also just to have better naming let's make the following changes:

`chunkCallback` -> `next`
`endCallback` -> `done`
`readChunked` -> `readChunk`

Reply:
Okay, on it